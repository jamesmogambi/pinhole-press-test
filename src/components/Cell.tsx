import React from "react";
import cellIcon from "../icon.svg";

type CellProp = {
  value: string | number;
  icon: boolean;
  styleProp: any;
};

const Cell: React.FC<CellProp> = ({ value, styleProp, icon }) => {
  return (
    <div className={`${styleProp} cell`}>
      {icon && <img src={cellIcon} alt="icon" className="icon" />}
      {value}
    </div>
  );
};

export default Cell;
