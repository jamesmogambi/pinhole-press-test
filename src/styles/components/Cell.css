.cell {
  width: 120px;
  font-size: 15px;
  padding: 10px;
  color: rgb(78, 78, 78);
  margin: 0px;
}

.cell-dark {
  background-color: var(--dark-gray-color);
}

.cell-light {
  background-color: var(--white-color);
  text-align: center;
}

.text-dark {
  color: #000000;
  font-weight: bold;
}

.text-light {
  color: var(--dark-gray-color);
}

.cell-double {
  width: 240px;
}

.cell-total {
  text-align: center;
}

.icon {
  height: 20px;
  margin-right: 10px;
}

.cell-icon {
  display: flex;
}
