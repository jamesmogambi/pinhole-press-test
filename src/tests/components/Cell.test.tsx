import { shallow } from "enzyme";
import Cell from "../../components/Cell";

test("should correctly render icon and category name ", () => {
  const wrapper = shallow(
    <Cell value="Furniture" styleProp="text-dark cell-icon" icon={true} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render subcategory name", () => {
  const wrapper = shallow(
    <Cell value="Bookcases" styleProp={null} icon={false} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render subtotal value", () => {
  const subtotal = 7550;
  const wrapper = shallow(
    <Cell value={subtotal} styleProp="cell-light" icon={false} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render total header", () => {
  const category = "Furniture";
  const wrapper = shallow(
    <Cell
      value={`${category} total`}
      styleProp="text-dark cell-double"
      icon={false}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render total value", () => {
  const total = 2590;
  const wrapper = shallow(
    <Cell
      value={total}
      styleProp="text-dark cell-dark cell-total"
      icon={false}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
