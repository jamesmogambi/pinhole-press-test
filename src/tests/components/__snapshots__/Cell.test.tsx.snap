// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render icon and category name  1`] = `
<div
  className="text-dark cell-icon cell"
>
  <img
    alt="icon"
    className="icon"
    src="icon.svg"
  />
  Furniture
</div>
`;

exports[`should correctly render subcategory name 1`] = `
<div
  className="null cell"
>
  Bookcases
</div>
`;

exports[`should correctly render subtotal value 1`] = `
<div
  className="cell-light cell"
>
  7550
</div>
`;

exports[`should correctly render total header 1`] = `
<div
  className="text-dark cell-double cell"
>
  Furniture total
</div>
`;

exports[`should correctly render total value 1`] = `
<div
  className="text-dark cell-dark cell-total cell"
>
  2590
</div>
`;
